File: src/runtime/dart/std/sys_native.dart
// std/sys_native.dart — Pytra Dart sys native seam
// source: src/runtime/dart/std/sys_native.dart

import 'dart:io' as _io;

final List<String> argv = <String>[];
final List<String> path = <String>[];
final dynamic stderr = _io.stderr;
final dynamic stdout = _io.stdout;

void exit([dynamic code]) {
  _io.exit(code is int ? code : 0);
}

void set_argv(List<dynamic> values) {
  argv.clear();
  argv.addAll(values.map((v) => v.toString()));
}

void set_path(List<dynamic> values) {
  path.clear();
  path.addAll(values.map((v) => v.toString()));
}

void write_stderr(dynamic text) {
  _io.stderr.write(text.toString());
}

void write_stdout(dynamic text) {
  _io.stdout.write(text.toString());
}

File: src/runtime/zig/std/sys_native.zig
const std = @import("std");
const pytra = @import("../built_in/py_runtime.zig");

var argv_store: pytra.Obj = pytra.empty_list();
var path_store: pytra.Obj = pytra.empty_list();

pub const argv: pytra.Obj = argv_store;
pub const path: pytra.Obj = path_store;
pub const stderr: []const u8 = "stderr";
pub const stdout: []const u8 = "stdout";

pub fn exit(code: i64) void {
    std.process.exit(@intCast(code));
}

pub fn set_argv(values: pytra.Obj) void {
    argv_store = values;
}

pub fn set_path(values: pytra.Obj) void {
    path_store = values;
}

pub fn write_stderr(text: []const u8) void {
    std.io.getStdErr().writer().writeAll(text) catch {};
}

pub fn write_stdout(text: []const u8) void {
    std.io.getStdOut().writer().writeAll(text) catch {};
}
